docs/architecture/adr-032-typed-events.md

```diff
@@ -0,0 +1,317 @@
+# ADR 032: Typed Events
+
+## Changelog
+
+- 28-Sept-2020: Initial Draft
+
+## Authors
+
+- Anil Kumar (@anilcse)
+- Jack Zampolin (@jackzampolin)
+- Adam Bozanich (@boz) 
+
+## Status
+
+Proposed
+
+## Abstract
+
+Currently in the SDK, events are defined in the handlers for each message, meaning each module doesn't have a cannonical set of types for each event. Above all else this makes these events difficult to consume as it requires a great deal of raw string matching and parsing. This proposal focuses on updating the events to use **typed events** defined in each module such that emiting and subscribing to events will be much easier. This workflow comes from the experience of the Akash Network team. 
```

Comment:
I still don't understand what this means :-/

What's a canonical event?

Wdym by events are defined in the handlers of each message? Events (types and attributes) are defined by the module and are emitted not only during message execution, but also during BeginBlock and EndBlock.

Reply:
The "canonical set of types" is referring to the proto definitions. Will add note about `BeginBlock` and `EndBlock`